### PR TITLE
Update to Python 3.12

### LINKED ISSUE #4848: Update to Python 3.12

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #4848

### CODE REVIEW

**Comment by MaxGhenis** [comment]
You'll need to update scipy as well

File: setup.py [hunk 1]
```diff
@@ -44,7 +45,7 @@
         "pytest-dependency",
         "pyyaml",
         "requests",
-        "scipy==1.10.1",
+        "scipy==1.14.0",
```
Comment (MaxGhenis) [inline setup.py:48]: If the current works
```suggestion
        "scipy<2",
```

**Comment by MaxGhenis** [comment]
Actually @anth-volk do you recall why you added scipy as a dep [here](https://github.com/PolicyEngine/policyengine-us/commit/0af066f8431a2532c64eb0afa960228b000b34f0)?

We do not currently import it anywhere.

**Comment by anth-volk** [comment]
@MaxGhenis This was due to a bug from about a year ago (I believe Oct. 2023) on deployment. For some reason, the `scipy` version at the time had to be pinned to prevent an upstream dependency error. It may be removable at this point.

File: setup.py [hunk 2]
```diff
@@ -44,7 +45,7 @@
         "pytest-dependency",
         "pyyaml",
         "requests",
-        "scipy==1.10.1",
+        "scipy<2",
```
Comment (MaxGhenis) [inline setup.py:48]:
```suggestion
```

**Review by MaxGhenis — APPROVED** [review]

**Comment by MaxGhenis** [comment]
OK I removed scipy

**Pull request merged**